auditlog/audit_log.go

```diff
@@ -0,0 +1,227 @@
+func (p EventParams) Equal(u convert.Equaler) bool {
+	other, ok := u.(EventParams)
+	if !ok {
+		return false
+	}
+	return reflect.DeepEqual(p, other)
+}
+
+// Value implements the driver.Valuer interface
+func (p EventParams) Value() (driver.Value, error) {
+	return toBytes(p)
+}
+
+// Scan implements the https://golang.org/pkg/database/sql/#Scanner interface
+// See also https://stackoverflow.com/a/25374979/835098
+// See also https://github.com/jinzhu/gorm/issues/302#issuecomment-80566841
+func (p *EventParams) Scan(src interface{}) error {
+	return fromBytes(src, p)
+}
+
+func toBytes(j interface{}) (driver.Value, error) {
+	if j == nil {
+		// log.Trace("returning null")
+		return nil, nil
+	}
+
+	res, error := json.Marshal(j)
+	return res, error
+}
+
+func fromBytes(src interface{}, target interface{}) error {
+	if src == nil {
+		target = nil
+		return nil
+	}
+	s, ok := src.([]byte)
+	if !ok {
+		return errs.Errorf("scanned source was not a string")
+	}
+	return json.Unmarshal(s, target)
+}
+
+// Repository provides functions to create and view audit logs
+type Repository interface {
+	Create(ctx context.Context, auditLog *AuditLog) error
+	LoadByID(ctx context.Context, id uuid.UUID) (AuditLog, error)
+	List(ctx context.Context, identityID uuid.UUID, start int, limit int) ([]AuditLog, int, error)
+}
+
+// NewRepository creates a GormRecordRepository
+func NewRepository(db *gorm.DB) Repository {
+	repository := &GormAuditLogRepository{
+		db: db,
+	}
+	return repository
+}
+
+// GormAuditLogRepository implements Repository using gorm
+type GormAuditLogRepository struct {
+	db *gorm.DB
+}
+
+// Create stores the given auditLog
+func (r *GormAuditLogRepository) Create(ctx context.Context, auditLog *AuditLog) error {
+	defer goa.MeasureSince([]string{"goa", "db", "auditLog", "create"}, time.Now())
+	// check values
+	if auditLog == nil {
+		return errors.NewBadParameterErrorFromString("missing audit log auditLog to persist")
+	}
+	if auditLog.IdentityID == uuid.Nil {
+		return errors.NewBadParameterError("identity_id", auditLog.IdentityID)
+	}
+	db := r.db.Create(auditLog)
+	if err := db.Error; err != nil {
+		return errors.NewInternalError(ctx, err)
+	}
+	return nil
+}
+
+// LoadByID returns the auditLog for the given id
+// returns NotFoundError or InternalError
+func (r *GormAuditLogRepository) LoadByID(ctx context.Context, id uuid.UUID) (AuditLog, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "auditLog", "loadById"}, time.Now())
+	result := AuditLog{}
+	tx := r.db.Model(AuditLog{}).Where("id = ?", id).First(&result)
+	if tx.RecordNotFound() {
+		log.Error(nil, map[string]interface{}{
+			"record_id": id,
+		}, "auditlog auditLog not found")
+		return AuditLog{}, errors.NewNotFoundError("auditlog_record", id.String())
+	}
+	if err := tx.Error; err != nil {
+		return AuditLog{}, errors.NewInternalError(ctx, err)
+	}
+	return result, nil
+}
+
+// List returns audit log records that belong to a given user
```

Comment:
Could you please update the comment to clarify what the return values of the function imply?

Reply:
yes, sure